### DISCUSSION TITLE
Wikilinks extension

### DISCUSSION THREAD

**Discussion by matterhorn103** [executablebooks/MyST-Parser#977] [Ideas]
Is there any chance of getting wikilink syntax support as mentioned in [#421](https://github.com/executablebooks/MyST-Parser/issues/421)?

Essentially as an optional extension that allows the wikilink syntax with the same meaning as the MyST cross-reference syntax.

Would surely be relatively easy to do? And would make inter-compatibility with Obsidian significantly better. I know Python so could help.

**Comment by chrisjsewell** [comment]
I think no

> And would make inter-compatibility with Obsidian significantly better

If obsidian compatibility is desired, then I feel a new parser/sphinx extension should be created.
Myst is its own syntax, it cannot be compatible with every other possible markdown flavor

**Comment by matterhorn103** [comment]
> Myst is its own syntax, it cannot be compatible with every other possible markdown flavor 

Of course and I agree, it just felt like a very simple thing.

Is it possible to write independent extensions that plug in to MyST-Parser the same way as with Sphinx extensions? I.e. would it be viable to write my own `myst-wikilinks` package and use it by adding it to the list of `myst_enable_extensions`?

**Reply by chrisjsewell** [reply]
> Is it possible to write independent extensions that plug in to MyST-Parser the same way as with Sphinx extensions?

No afraid not

> it just felt like a very simple thing.

it is really not simple at all 😅; you need to implement a parser that covers all of https://help.obsidian.md/Editing+and+formatting/Obsidian+Flavored+Markdown#Supported+Markdown+extensions and only these.

Otherwise, you will just end up with a mess of different syntaxes that do not really compatible with anything

**Reply by matterhorn103** [reply]
> you need to implement a parser that covers all of https://help.obsidian.md/Editing+and+formatting/Obsidian+Flavored+Markdown#Supported+Markdown+extensions and only these.

No, I'm not suggesting covering the entire set of extensions that Obsidian uses. That wouldn't be necessary. If users want to write Markdown that's compatible with both Obsidian and MyST, they just have to stick to the subset of syntaxes that are compatible with both (which is a fair amount i.e. GFM + strikethrough + math + footnotes).

Indeed there are already people who do exactly that and use Obsidian as an environment for writing Markdown input for a static site generator.

I'm literally just suggesting parsing `[[some/link|text]]` as `[text](some/link)` (probably while ignoring images that have the syntax `![[some/image|size]]`).
